### PR TITLE
fix(material/core): Move remaining MDC core functionality out of experimental

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/e2e-app/theme.scss [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 // **Be sure that you only ever include this mixin once!**
 @include mat.core();
-@include experimental.all-mdc-component-typographies();
 
 // Define the default theme (same as the example above).
 $candy-app-primary: mat.define-palette(mat.$indigo-palette);
 $candy-app-accent: mat.define-palette(mat.$pink-palette, A200, A100, A400);
-$candy-app-theme: mat.define-light-theme($candy-app-primary, $candy-app-accent);
+$candy-app-theme: mat.define-light-theme((
+    color: (primary: $candy-app-primary, accent: $candy-app-accent),
```
Comment (crisbeto) [inline src/e2e-app/theme.scss:14]: Indentation seems off here.

File: src/material/core/typography/_all-typography.scss [hunk 2]
```diff
@@ -33,11 +35,82 @@
...
+/// @param {Map} $headline-6 The font settings for the headline-6 font level.
+/// @param {Map} $subtitle-1 The font settings for the subtitle-1 font level.
+/// @param {Map} $subtitle-2 The font settings for the subtitle-2 font level.
+/// @param {Map} $body-1 The font settings for the body-1 font level.
+/// @param {Map} $body-2 The font settings for the body-2 font level.
+/// @param {Map} $caption The font settings for the caption font level.
+/// @param {Map} $button The font settings for the button font level.
+/// @param {Map} $overline The font settings for the overline font level.
+/// @return {Map} A map containing font settings for each of the levels in the Material Design spec.
+@function define-mdc-typography-config(
```
Comment (crisbeto) [inline src/material/core/typography/_all-typography.scss:69]: Should we merge this into `define-typography-config` instead?

File: src/universal-app/theme.scss [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
 // **Be sure that you only ever include this mixin once!**
 @include mat.core();
-@include experimental.all-mdc-component-typographies();
 
 // Define the default theme (same as the example above).
 $candy-app-primary: mat.define-palette(mat.$indigo-palette);
 $candy-app-accent: mat.define-palette(mat.$pink-palette, A200, A100, A400);
-$candy-app-theme: mat.define-light-theme($candy-app-primary, $candy-app-accent);
+$candy-app-theme: mat.define-light-theme((
+    color: (primary: $candy-app-primary, accent: $candy-app-accent),
```
Comment (crisbeto) [inline src/universal-app/theme.scss:14]: Indentation here as well.

File: src/e2e-app/theme.scss (hunk 1 above)
Comment (mmalerba) [reply src/e2e-app/theme.scss:14]: what's off about it? it looks fine to me, and yarn format isn't complaining

File: src/material/core/typography/_all-typography.scss (hunk 2 above)
Comment (mmalerba) [reply src/material/core/typography/_all-typography.scss:69]: Oh thanks for pointing that out, I wanted to have a discussion about this. I'll post on our team chat

File: src/universal-app/theme.scss (hunk 3 above)
Comment (mmalerba) [reply src/universal-app/theme.scss:14]: looks fine to me

File: src/material/core/typography/_all-typography.scss (hunk 2 above)
Comment (mmalerba) [reply src/material/core/typography/_all-typography.scss:69]: Based on our discussion, I'll do a followup PR where I rename this and add in a way for people who need `px` units to more easily switch to this function. I added a TODO for now

File: src/e2e-app/theme.scss (hunk 1 above)
Comment (crisbeto) [reply src/e2e-app/theme.scss:14]: Maybe it's just me, but it seems like it's twice as indented as it should be. Either it's using 4 spaces or 2 tabs.

**Review by crisbeto — APPROVED** [review]
LGTM

**Pull request merged**